core/src/services/hdfs/backend.rs

```diff
@@ -277,8 +277,9 @@ impl Access for HdfsBackend {
     async fn read(&self, path: &str, args: OpRead) -> Result<(RpRead, Self::Reader)> {
         let p = build_rooted_abs_path(&self.root, path);
```

Comment:
Maybe we can move parts of this implementation to core as `HdfsCore::hdfs_read_from_path`. Once we have done that, the `read` function will become:

1. read the file
2. return results (`HdfsReader` is what OpenDAL returns for users to read data from)